sdcm/db_log_reader.py

```diff
@@ -98,6 +98,8 @@ def _read_and_publish_events(self) -> None:
                     self._skipped_end_line += 1
                     continue
                 try:
+                    if len(line) > 16384:
```

Comment:
Maybe it is better to have line size limit for each of the regexes?
Set 16k by default and allow to use much smaller sizes for short messages.
It will minimize the number of regex match operations a lot and replace it with fast condition checks.

Reply:
it's a bit more problematic, I don't want to searches, and I don't want to print those in SCT logs
and there are the backtraces regexes, that it would need to skip as well

basically I want to ignore those big one, and not start thinking in each regexes 

now I would experiment to see which size is affecting the worst on the CPU. so I'll be able to pick the smallest number possible here.

Reply:
best would be to peek into the machine during high load and see which process takes so much. I'm not sure of the mechanism Python would take more than 1cpu due to GIL and how it works, so unless we don't match regexps in separate processes, possibly some other process takes the rest 3 cpus.

Reply:
You can completely remove backtrace detection of old format (larger than one row).
If there is such, we will open an issue to change it to one liner.

Reply:
we already has such an issue, years ago, but reactor stalls weren't changed.

Reply:
Can you please point me to this issue? (I recall one, but I thought we changed everything).

You can proceed with that, I'll add my comment there that it's going to be deprecated.

Reply:
I've raise few years back,
https://github.com/scylladb/scylladb/issues/5464

but not everything was changed, just a small part

Reply:
This one is closed.
Whatever wasn't fixed require new issue.
If we know that reactor stalls are still like that, please open a new issue for it.
I'll alert there that soon SCT will ignore reactor stalls backtraces and won't decode them if it's not fixed.

Reply:
> best would be to peek into the machine during high load and see which process takes so much. I'm not sure of the mechanism Python would take more than 1cpu due to GIL and how it works, so unless we don't match regexps in separate processes, possibly some other process takes the rest 3 cpus.

we have a separate subprocess for each node we are tracking that doesn't the reading and regexes, so if multiple nodes generate those big lines it would hit multiple CPUs

the limit was proven to help, I'm gonna take it even to a lower number, like 5K, we don't want try match any regexes on a big prints, they won't be of any interest to us (compactio/resharding/bootstrap), we are mostly looking for errors level print, aborts, segfaults, and that should be our focus when generating events.

the rest of the code that follow a file for specific regexs, doesn't have this logs, and would chock a bit on those big line, but it usually has one or two regexes to match